### PR TITLE
fix(pipelinetemplate): Child stages of conditional stages should be p…

### PR DESCRIPTION
**Pull request by jeyrschabu**
…reserved

- Fixed an issue where children of a rendered conditional stage are dropped off the stages graph
- If s2 is a conditional stage then:
Input:
 s1 <- s2 <- s4
       \- s3 

Output:
   s1 <- s4
	  \- s3

### CODE REVIEW

**Comment by jeyrschabu** [comment]
@spinnaker/netflix-reviewers PTAL

File: orca-pipelinetemplate/src/main/java/com/netflix/spinnaker/orca/pipelinetemplate/v1schema/graph/transform/ConditionalStanzaTransform.java [hunk 1]
```diff
@@ -46,25 +47,18 @@ public void visitPipelineTemplate(PipelineTemplate pipelineTemplate) {
...
+      .stream()
+      .filter(stage -> stage.getWhen() != null && !stage.getWhen().isEmpty())
+      .forEach(stage -> {
+        RenderContext context = new DefaultRenderContext(templateConfiguration.getPipeline().getApplication(), template, trigger);
+        context.getVariables().putAll(templateConfiguration.getPipeline().getVariables());
+        for (String conditional : stage.getWhen()) {
+          String rendered = renderer.render(conditional, context);
+          if (!Boolean.parseBoolean(rendered)) {
+            stage.setConditional(true);
+          }
```
Comment (robzienert) [inline orca-pipelinetemplate/src/main/java/com/netflix/spinnaker/orca/pipelinetemplate/v1schema/graph/transform/ConditionalStanzaTransform.java]: This logic reads weird. I see why you wrote it the way you did, but looking at this conditional on its own (without seeing `ConfigStageInjectionTransform`), you'd think it was logically a bug. Definitely warrants a comment like you have below.

**Comment by robzienert** [comment]
🥇

File: orca-pipelinetemplate/src/main/java/com/netflix/spinnaker/orca/pipelinetemplate/v1schema/graph/transform/ConditionalStanzaTransform.java (hunk 1 above)
Comment (jeyrschabu) [inline orca-pipelinetemplate/src/main/java/com/netflix/spinnaker/orca/pipelinetemplate/v1schema/graph/transform/ConditionalStanzaTransform.java]: Will rename the method.

**Pull request merged**